pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ActivityInstanceDao.java

```diff
@@ -202,6 +202,21 @@ int reassignInstancesInStudy(
             + " limit 1")
     Optional<Long> findMostRecentInstanceBeforeCurrent(@Bind("instanceId") long currentInstanceId);
 
+    @SqlQuery("select count(ai.activity_instance_id)"
```

Comment:
Wondering if we really need to do this for every activity ? 
May be to improve overall performance we can avoid executing this query for activities which does not support more than 1 activity instance.
For example: look for study activities  with study_activity.max_instances_per_user != 1 and execute this sql for only those activities.
We barely have less than 5 or 10% of activities which have sa.max_instances_per_user = null.
select * from study_activity sa where sa.max_instances_per_user is null or sa.max_instances_per_user > 1
-- may be just sa.max_instances_per_user != 1

I could be overlooking something though.

Reply:
I can add in a similar optimization here!

Reply:
I've put in an optimization that addresses this!

Reply:
thank you